internal/bminventory/inventory.go

```diff
@@ -316,6 +316,18 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 	id := strfmt.UUID(uuid.New().String())
 	url := installer.GetClusterURL{ClusterID: id}
 	log.Infof("Register cluster: %s with id %s", swag.StringValue(params.NewClusterParams.Name), id)
+	success := false
+	defer func() {
+		if success {
```

Comment:
can you make it a function? i mean start from "if success..". Then i can use in day2 cluster

Reply:
Do you want me to add it to `RegisterAddHostsCluster` shouldn't it have a different message?

Reply:
we can make the message as a parameter. No need to add it to day2 cluster, we will be doing it shortly anyway

Reply:
Isn't it a bit overkill? the call will looks something like this
```
	defer func() {
		HandleRegisterErrorWithEvent(success, "message for success", "message for error")
	}()
```
It's not some generic function because it does not have an event in case of error because the user will not have any id to look for this error

Reply:
I don't mind to add this function, just asking because it's not something generic.

Reply:
from my point of view it is not a must, i just think it will look better, since we won't see 10 lines of almost identical code in 2 consecutive functions. you decide :)